graphs/possible_bipartition.py

```diff
@@ -1,12 +1,38 @@
+
 # Can be used for BFS
-from collections import deque 
+from collections import deque
+# Check whether Graph is Bipartite or Not using DFS
+
+# A Bipartite Graph is a graph whose vertices can be divided into two independent sets,
+# U and V such that every edge (u, v) either connects a vertex from U to V or a vertex
+# from V to U. In other words, for every edge (u, v), either u belongs to U and v to V,
+# or u belongs to V and v to U. We can also say that there is no edge that connects
+# vertices of same set.
 
 def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)
```

Comment:
This works, but the time complexity would be O(NE) because of the loops in lines 33-36.

This can be done using DFS in O(N + E) time, but you would need to modify your `dfs` helper function and how you handle the if statement in lines 30-31.

Otherwise nice `dfs` solution.